src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -431,9 +431,15 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
         case PFPredBase(id) =>
           val idT = idType(id)
           PredT(idT.asInstanceOf[FunctionT].args.zip(args).collect{ case (typ, None) => typ })
-        case p: PMPredBase =>
-          val idT = idType(p.id)
-          PredT(idT.asInstanceOf[FunctionT].args.tail.zip(args).collect{ case (typ, None) => typ })
+        case p: PDottedBase => resolve(p.recvWithId) match {
+          case Some(_: ap.Predicate) =>
+            val recvWithIdT = exprOrTypeType(p.recvWithId)
+            PredT(recvWithIdT.asInstanceOf[FunctionT].args.zip(args).collect{ case (typ, None) => typ })
+          case Some(_: ap.PredicateExpr) => ??? // Predicate expressions are not supported at the moment
+          case _ =>
+            val idT = idType(p.id)
+            PredT(idT.asInstanceOf[FunctionT].args.tail.zip(args).collect{ case (typ, None) => typ })
```

Comment:
What case is this supposed to be? That just looks like the case where the base is a predicate. In my opinion, there should be a violation here stating that an unexpected case was encountered.

Reply:
I think the case that is missing is the received predicate, but this should also be added as a pattern match with an `AstPattern`. For this also the implementation is wrong, I think. You should not take the type of the id, but the type of the entire base `p.recvWithId`. Otherwise, you do not account for the receiver of the mpredicate.